examples/cute/tutorial/xe_gemm.cpp

```diff
@@ -0,0 +1,580 @@
+  else if constexpr (cute::is_same_v<TA, cute::bfloat16_t>)
+    return XE_DPAS_TT<8, float, cute::bfloat16_t>{};
+  else
+    return XE_DPAS_TT<8, float, cute::half_t>{};
+}
+
+template <class ATensor, class BTensor, class CTensor>
+auto
+choose_tiled_mma(ATensor const& A, BTensor const& B, CTensor const&)
+{
+  using TA = typename ATensor::element_type;
+  using TB = typename BTensor::element_type;
+  using TC = typename CTensor::element_type;
+
+  auto op = choose_mma_op<TA,TB,TC>();
+
+  constexpr bool byte = (cute::max(sizeof_bits_v<TA>, sizeof_bits_v<TB>) <= 8);
+  constexpr bool use_1x_dpas = is_constant_v<1, decltype(stride<0>(A))>             // Use single-DPAS chunks for A^T case
+                            || (byte && is_constant_v<1, decltype(stride<0>(B))>);  //  pending compiler improvements (also int8 B^N)
+
+  using _K = conditional_t<use_1x_dpas,
+                           C<op.K>, C<op.K*2>>;
+
+  using WGTile = Shape<_256, _256, _K>;                               // 256x256 WG tile size
+  using SGLayout = Layout<Shape<_8, _4, _1>, Stride<_4, _1, _0>>;     // 8x4 SG tiling, n-major
+
+  using MMA = typename TiledMMAHelper<MMA_Atom<decltype(op)>, Layout<WGTile>, SGLayout>::TiledMMA;
+
+  return MMA{};
+}
+
+template <class ATensor, class BTensor, class CTensor>
+void
+gemm_cute(sycl::queue &Q,
+          ATensor   const& A,         // (M,K)
+          BTensor   const& B,         // (N,K)
+          CTensor        & C)         // (M,N)
+{
+  auto mma = choose_tiled_mma(A, B, C);
+
+  sycl::range<2> local = {size(mma), 1};
+  sycl::range<2> global = {local[0] * ceil_div(shape<0>(B), get<1>(mma.tile_mnk())),
+                           local[1] * ceil_div(shape<0>(A), get<0>(mma.tile_mnk()))};
+
+  namespace syclex = sycl::ext::oneapi::experimental;
+  namespace intelex = sycl::ext::intel::experimental;
+
+  syclex::properties kernel_props {
+    syclex::sub_group_size<16>,
+    intelex::grf_size<256>
+  };
+
+  auto event = Q.parallel_for(sycl::nd_range<2>(global, local), kernel_props,
+    [=](auto) {
+      gemm_device(A, B, C, mma);
+    }
+  );
+
+  EventManager::getInstance().addEvent(event);
+}
+
+template <typename T>
+auto ensure_signed_helper_t() {
+  if constexpr (is_unsigned_v<T>)
+    return make_signed_t<T>{};
+  else
+    return T{};
+}
+
+template <typename T>
+using ensure_signed_t = decltype(ensure_signed_helper_t<T>());
+
+template <class ATensor, class BTensor, class CTensor>
+bool
+gemm_verify(sycl::queue &Q,
+            ATensor const& A,         // (M,K)
+            BTensor const& B,         // (N,K)
+            CTensor const& C)         // (M,N)
+{
+  int m = size<0>(A);
+  int n = size<0>(B);
+  int k = size<1>(A);
+
+  auto ok = sycl::malloc_shared<bool>(1, Q);
+  *ok = true;
+
+  Q.parallel_for(sycl::range<2>(m, n), [=](sycl::item<2> id) {
+    int i = id[0], j = id[1];
+
+    using AccType = typename CTensor::element_type;
+    using SignedAccType = ensure_signed_t<AccType>;
```

Comment:
```suggestion
    using SignedAccType = make_signed_t<AccType>;
```
I don't think `ensure_signed_t` is needed. make_signed works fine if it is already signed.

Reply:
The problem is that `std::make_signed` is only valid for integer types, so it needs this wrapper to handle float types.

Reply:
makes sense